app/models/assignment.rb

```diff
@@ -649,40 +651,17 @@ def replace_submission_rule(new_submission_rule)
   def next_criterion_position
     # We're using count here because this fires off a DB query, thus
     # grabbing the most up-to-date count of the criteria.
-    get_criteria.count > 0 ? get_criteria.last.position + 1 : 1
+    criteria.count > 0 ? criteria.last.position + 1 : 1
   end
 
   # Returns a filtered list of criteria.
   def get_criteria(user_visibility = :all, type = :all, options = {})
-    @criteria ||= Hash.new
-    unless @criteria[[user_visibility, type, options]].nil? || options[:no_cache]
-      return @criteria[[user_visibility, type, options]]
-    end
-
-    include_opt = options[:includes]
-    if user_visibility == :all
-      @criteria[[user_visibility, type, options]] = get_all_criteria(include_opt)
-    elsif user_visibility == :ta
-      @criteria[[user_visibility, type, options]] = get_ta_visible_criteria(include_opt)
-    elsif user_visibility == :peer
-      @criteria[[user_visibility, type, options]] = get_peer_visible_criteria(include_opt)
-    end
-  end
-
-  def get_all_criteria(include_opt)
-    criteria.includes(include_opt).order(:position)
-  end
-
-  def get_ta_visible_criteria(include_opt)
-    get_all_criteria(include_opt).select(&:ta_visible)
-  end
-
-  def get_peer_visible_criteria(include_opt)
-    get_all_criteria(include_opt).select(&:peer_visible)
+    # can't use select(&:all) because all isn't a field
+    criteria.includes(options[:includes]).order(:position).select(&user_visibility)
   end
 
   def criteria_count
-    get_criteria.size
+    criteria.size
```

Comment:
Actually, can you please remove this method, and replace instances of `criteria_count` with `criteria.size`?

Reply:
`criteria_count` actually wasn't used in the codebase